File: WEBSITE/trader-os-companion-main/together_affirmation.py
import requests

api_key = "0097473bda6ba5d1b430afc0cefdcbd87820c51bbd6a64d6ce6cb33cc1ee9acc"
chat_endpoint = "https://api.together.xyz/v1/chat/completions"
moderation_endpoint = "https://api.together.xyz/v1/completions"

headers = {
    "Authorization": f"Bearer {api_key}",
    "Content-Type": "application/json"
}

def generate_affirmation(issue):
    payload = {
        "model": "meta-llama/Llama-3.3-70B-Instruct-Turbo",
        "messages": [
            {"role": "system", "content": "You are a mindset coach for traders. Generate a short, calming affirmation for the issue below."},
            {"role": "user", "content": f"My issue: {issue}"}
        ],
        "temperature": 0.7,
        "max_tokens": 100
    }
    try:
        response = requests.post(chat_endpoint, headers=headers, json=payload)
        response.raise_for_status()
        data = response.json()
        return data["choices"][0]["message"]["content"].strip()
    except Exception as e:
        print("[Affirmation API Error]", e)
        if hasattr(e, 'response') and e.response is not None:
            print(e.response.text)
        return None

def moderate_text(text):
    payload = {
        "model": "meta-llama/Llama-Guard-4-12B",
        "prompt": text,
        "max_tokens": 200,
        "temperature": 0
    }
    try:
        response = requests.post(moderation_endpoint, headers=headers, json=payload)
        response.raise_for_status()
        data = response.json()
        output = data["choices"][0]["text"]
        return output
    except Exception as e:
        print("[Moderation API Error]", e)
        if hasattr(e, 'response') and e.response is not None:
            print(e.response.text)
        return None

def get_safe_affirmation():
    issue = input("Describe your trading mindset issue: ")
    affirmation = generate_affirmation(issue)
    if not affirmation:
        print("Failed to generate affirmation.")
        return
    moderation_result = moderate_text(affirmation)
    if moderation_result and 'Unsafe: True' in moderation_result:
        print("This affirmation was flagged as unsafe. Try a different prompt.")
    elif moderation_result:
        print("\nYour personalized affirmation:\n")
        print(affirmation)
    else:
        print("Moderation failed. Could not verify safety.")

if __name__ == "__main__":
    get_safe_affirmation()

File: WEBSITE/mind2profit-companion-main/src/api/togetherAffirmation.js
const express = require('express');
const router = express.Router();
const axios = require('axios');
const fs = require('fs');
const path = require('path');

const TOGETHER_API_KEY = "0097473bda6ba5d1b430afc0cefdcbd87820c51bbd6a64d6ce6cb33cc1ee9acc";

const styles = [
  "Make it poetic.",
  "Make it sound like a sports coach.",
  "Make it gentle and nurturing.",
  "Make it sound like a wise old mentor.",
  "Make it short and punchy.",
  "Use a metaphor related to nature."
];

function getRandomStyle() {
  return styles[Math.floor(Math.random() * styles.length)];
}

router.post('/', async (req, res) => {
  const { challenge, behavior, tone, routine } = req.body;

  try {
    const randomStyle = getRandomStyle();
    const systemPrompt = `
You are a creative and insightful mindset coach for traders.
Each time, generate a truly unique, highly personalized affirmation that is different in style, structure, and language from previous ones.
Vary the sentence structure, use metaphors, analogies, or different tones (gentle, motivational, poetic, etc).
Never use the same sentence structure or opening lines as previous affirmations. Avoid repeating phrases, and do not use the phrase "I am confident and capable", "I wait for my confirmations with ease and grace", or similar templates. Start each affirmation in a completely new way. Do not start with "I am" or "You are". Begin with a metaphor, a question, or a vivid image.
Draw on the user's details to make each affirmation feel fresh and original.
`;
    const userPrompt = `
I'm a trader facing the following challenge: ${challenge}.
I'd like to work on this behavior: ${behavior}.
I prefer the affirmation to have a ${tone} tone and to be used as a ${routine}.
${randomStyle}
Please surprise me with something original and different from what you've written before.
`;

    const response = await axios.post(
      'https://api.together.xyz/v1/chat/completions',
      {
        model: "deepseek-ai/DeepSeek-R1-Distill-Llama-70B-free",
        messages: [
          { role: "system", content: systemPrompt },
          { role: "user", content: userPrompt }
        ],
        temperature: 0.9,
        max_tokens: 120
      },
      {
        headers: {
          "Authorization": `Bearer ${TOGETHER_API_KEY}`,
          "Content-Type": "application/json"
        }
      }
    );

    const outputPath = path.join(__dirname, '../../data/affirmation_log.json');
    const outputData = {
      timestamp: new Date().toISOString(),
      challenge,
      behavior,
      tone,
      routine,
      style: randomStyle,
      affirmation: response.data.choices[0].message.content
    };
    fs.writeFileSync(outputPath, JSON.stringify(outputData, null, 2));

    res.json({ affirmation: response.data.choices[0].message.content });
  } catch (err) {
    res.status(500).json({ error: err.message, details: err.response?.data });
  }
});

module.exports = router;
